### PR TITLE
#1042-project-detail-page-revamp-skeleton

### LINKED ISSUE #1042: Project Detail Page Revamp - Skeleton
Make a new feature branch for this feature, then make a PR off of that branch to merge in this work. Ask in #software if that doesn't mean anything to you.

Make a new file for the project detail page that looks the same as before, but instead of all the data and page blocks, it has 5 tabs 'Overview' 'Tasks' 'Scope' 'Gantt' and 'Changes'. These tabs shouldn't be populated yet, they'll be filled out in later tickets.

See TasksLists code for how to do tabs
image

### PR DESCRIPTION
**Pull request by mouri-e**
## Changes

Added file "ProjectDetailTabs.tsx" to make the 5 tabs as requested on the project details page. Changed the folder structure to separate the new file and old files for the project details page. Edited "ProjectViewContainer.tsx" so only the requested tabs would be displayed on the project details page.

## Notes

-N/A

## Test Cases

-N/A

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

image
image

## To Do

_Any remaining things that need to get done_

- [x] File Organization/name changes
- [ ] Possible code changes

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1042

### CODE REVIEW

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { Project, wbsPipe } from 'shared';
+import { routes } from '../../../../utils/routes';
+
+interface ProjectDetailPageTabProps {
+  project: Project;
+}
+
+// Page block containing task list view
+const ProjectDetailTabs = ({ project }: ProjectDetailPageTabProps) => {
+  // Values that go in the URL depending on the tab value, example /projects/0.0.0/in-progress
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx:18]: this comment is not accurate

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { useEffect, useMemo, useState } from 'react';
+import { Link, useLocation, useRouteMatch } from 'react-router-dom';
+import { Project, wbsPipe } from 'shared';
+import { routes } from '../../../../utils/routes';
+
+interface ProjectDetailPageTabProps {
+  project: Project;
+}
+
+// Page block containing task list view
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx:16]: this comment is not accurate

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Tab, Tabs } from '@mui/material';
+import { useEffect, useMemo, useState } from 'react';
+import { Link, useLocation, useRouteMatch } from 'react-router-dom';
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx:8]: we rename all links from react-router-dom to RouterLink because it conflicts with the mui links

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListComponents.tsx [hunk 4]
```diff
@@ -71,20 +71,5 @@ export const AssigneeEdit = (params: GridRenderEditCellParams) => {
...
-      options={options}
-      getOptionLabel={(option) => option.label}
-      onChange={handleValueChange}
-      value={assignees.map((u: UserPreview) => options.find((o: any) => o.id === u.userId)!)}
-      // TODO: make assignees an array with a custom method
-      renderInput={(params) => <TextField {...params} variant="outlined" placeholder="Select A User" />}
-      ref={handleRef}
-    />
-  );
+  return <></>;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListComponents.tsx:74]: why did you do this?

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx [hunk 5]
```diff
@@ -306,22 +306,7 @@ const TaskListDataGrid: React.FC<TaskListDataGridProps> = ({
...
-      pageSize={pageSize}
-      rowsPerPageOptions={[5, 10, 15, 100]}
-      onPageSizeChange={(newPageSize) => {
-        localStorage.setItem(tableRowCount, String(newPageSize));
-        setPageSize(newPageSize);
-      }}
-      sx={styles.datagrid}
-    />
-  );
+  return <></>;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx:309]: why did you change these?

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
UI looks good but why did you rename it to old project view container, you can just leave them in project view container and well update it as we go along with this feature branch. Also Remember to tag your commits

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListComponents.tsx (hunk 4 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListComponents.tsx:74]: This was a misunderstanding when I read the ticket, I changed it back to the original.

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx (hunk 5 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx:309]: Also a misunderstanding, changed it back.

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx (hunk 2 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx:16]: Changed the comment to be relevant to the newly created tabs.

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx (hunk 3 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx:8]: done

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx [hunk 6]
```diff
@@ -9,13 +9,9 @@ import { wbsPipe } from '../../../utils/pipes';
 import { useAuth } from '../../../hooks/auth.hooks';
 import ChangesList from '../../../components/ChangesList';
 import DescriptionList from '../../../components/DescriptionList';
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx:11]: make sure to remove any unused imports like this one

**Review by RChandler234 — CHANGES_REQUESTED** [review]
you don't need an oldprojectview folder you can just leave all those files where they are and we'll delete the unused ones when we merge the feature branch. you also don't need to add in a project folder under the ProjectViewContainer file

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx (hunk 1 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/Project/ProjectDetailTabs.tsx:18]: I changed it to be "Scope" instead of "in-progress".

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx (hunk 6 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx]: Removed all unused imports in "ProjectViewContainer.tsx" and "ProjectDetailTabs.tsx" and checked all other files in the parent folder for unused imports.

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  // Values that go in the URL depending on the tab value, example /projects/0.0.0/Scope
+  const tabUrlValues = useMemo(() => ['Overview', 'Tasks', 'Scope', 'Gantt', 'Changes'], []);
+
+  const match = useRouteMatch<{ wbsNum: string; tabValueString: string }>(`${routes.PROJECTS}/:wbsNum/:tabValueString`);
+  const tabValueString = match?.params?.tabValueString;
+  const wbsNum = wbsPipe(project.wbsNum);
+
+  // Default to the "overview" tab
+  const initialValue: number = tabUrlValues.indexOf(tabValueString ?? 'Overview');
+  const [value, setValue] = useState<number>(initialValue);
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx:27]: rename to tab

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+const ProjectDetailTabs = ({ project }: ProjectDetailPageTabProps) => {
+  // Values that go in the URL depending on the tab value, example /projects/0.0.0/Scope
+  const tabUrlValues = useMemo(() => ['Overview', 'Tasks', 'Scope', 'Gantt', 'Changes'], []);
+
+  const match = useRouteMatch<{ wbsNum: string; tabValueString: string }>(`${routes.PROJECTS}/:wbsNum/:tabValueString`);
+  const tabValueString = match?.params?.tabValueString;
+  const wbsNum = wbsPipe(project.wbsNum);
+
+  // Default to the "overview" tab
+  const initialValue: number = tabUrlValues.indexOf(tabValueString ?? 'Overview');
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx:26]: rename to initialTab

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx (hunk 7 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx:27]: or tabValue

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListTabPanel.tsx [hunk 9]
```diff
@@ -19,7 +19,7 @@ import {
 import { useToast } from '../../../../hooks/toasts.hooks';
 import { TaskListTabPanelProps, transformDate } from '../../../../utils/task.utils';
 import ErrorPage from '../../../ErrorPage';
-import TaskListDataGrid from './TaskListDataGrid';
+import TaskListDataGrid from '../TaskList/TaskListDataGrid';
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListTabPanel.tsx:22]: why did u change this?

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
looks good just some nit picky things

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx (hunk 7 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx:27]: renamed

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListTabPanel.tsx (hunk 9 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListTabPanel.tsx:22]: changed back

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx (hunk 8 above)
Comment (mouri-e) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetailTabs.tsx:26]: renamed
